browser/src/components/core/Editor.tsx

```diff
@@ -0,0 +1,244 @@
+          editor.getHTML()
+        )
+      );
+    },
+  })
+
+  const setLink = (cancel: boolean) => {
+    if (cancel) {
+      // Previous url
+      return editor.getAttributes('link').href;
+    }
+    var url = linkInput;
+
+    if (url === "" || url === null || url === undefined) {
+      //editor.chain().focus().extendMarkRange('link').unsetLink().run()
+      editor.chain().focus().unsetLink().run()
+      return
+    }
+    
+    // Add so href doesn't point to pluriverse.world/{url}  
+    if (
+      !(url.toLowerCase().startsWith("https://") 
+      || url.toLowerCase().startsWith("http://"))
+    ) {
+      // could also add //, but then displays as //google.com in input
+      // which is potentially confusing
+      url = "http://" + url;    
+    }
+
+    // Update link
+    editor.chain().focus().setLink({ href: url }).run()
+//editor.chain().focus().extendMarkRange('link').setLink({ href: url }).run()
+  };
+
+  const onClickAddLink = () => {
+    setLink(false);
+    setDisplayLinkModal(false);
+    setLinkInput(null);
+    toggleBackgroundScrollingOnModal(true)
+  };
+
+  const onCloseModal = () => {
+    setLink(true);
+    setDisplayLinkModal(false);
+    setLinkInput(null);
+    toggleBackgroundScrollingOnModal(true);
+  };
+
+  const getPreviousLink = () => {
+    const prevUrl = editor.getAttributes('link').href;
+    setLinkInput(prevUrl);
+  }
+
+  const toggleBackgroundScrollingOnModal = (scrollable: boolean) => {
+    if (scrollable) {
+      document.body.style.overflow = "scroll";
+    } else {
+      document.body.style.overflow = "hidden";
+    }
+  }
+
+  const handleKeyPress = (event) => {
+    // Enter key press
+    if (event.key === "Enter") {
+      onClickAddLink();
+    }
+  }
+
+  return (
+    <>
+      {editor &&
+        <>
+          <div className="menu">
+            <button
+              onClick={() => editor.chain().focus().toggleBold().run()}
+              className={`menuItem ${editor.isActive('bold') ? 'shimmer' : ''}`}
+            >
+              <strong>B</strong> 
+            </button>
+            <button
+              onClick={() => editor.chain().focus().toggleItalic().run()}
+              className={`menuItem ${editor.isActive('italic') ? 'shimmer' : ''}`}
+            >
+              <em>I</em> 
+            </button>
+            <button
+              onClick={() => {
+                openModal();
+              }}
+              className={`menuItem linkIcon ${displayLinkModal ? 'shimmer' : 'white'}`}
+            >
+              <strong>🔗</strong> 
+            </button>
+          </div>
+        </>
+      }
+      <Modal 
+        isOpen={displayLinkModal} 
+        onAfterOpen={getPreviousLink}
+        className="modal" 
+        overlayClassName="overlay"
+        onRequestClose={() => onCloseModal()}
+        shouldCloseOnOverlayClick={true}
+      >
+        <h3 className="text-3xl font-bold">
+          Add Link 
+        </h3>
+        <div>
+          <input 
```

Comment:
just checking is there any sort of validation on this to make sure it's a valid link? if not some simple validation would be nice (i think you can find a regex online)

Reply:
oops, super obvious thing to leave out -- will fix tomorrow 🙏